File: 0611-valid-triangle-number/Solution.go
package Leetcode

import "sort"

func triangleNumber(nums []int) int {
	sort.Ints(nums)
	count := 0

	for i := 2; i < len(nums); i++ {
		left, right := 0, i-1
		for left < right {
			if nums[left]+nums[right] > nums[i] {
				count += right - left
				right--
			} else {
				left++
			}
		}
	}
	return count
}

File: 0611-valid-triangle-number/Solution.cs
public class Solution {
    public int TriangleNumber(int[] nums)
    {
        Array.Sort(nums);
        int count = 0;
        for (int i = 2; i < nums.Length; ++i)
        {
            int left = 0;
            int right = i - 1;
            while (left < right)
            {
                if (nums[left] + nums[right] > nums[i])
                {
                    count += right - left;
                    right--;
                }
                else
                {
                    left++;
                }
            }
        }

        return count;
    }

}
